streetcrm/admin.py

```diff
@@ -577,6 +577,39 @@ def export_search(request, form, advanced, search_query):
 
         return response
 
+    # This is based on export_search, but is different enough that abstracting out the common
+    # underlying looked to be too great an effort.  That could be reworked to collapse them well
+    @staticmethod
+    def export_results(csv_writer, header, results):
+        """
+        Export the set of search results in the given CSV.
+        """
+        from streetcrm import models
+
+        csv_writer.writerow([ col['heading'] for col in header ])
+
+        for result in results:
+            if result is None:
+                continue
+
+            new_row = []
+            # Special columns include: institution_id, leadership
+            for col in header:
+                # Display names of related objects, not IDs
+                if col['column'] == 'institution_id' and result.__dict__[col['column']] != None:
+                    this_inst = models.Institution.objects.get(id=result.__dict__[col['column']])
+                    new_row.append(this_inst.name)
+                elif col['column'] == 'leadership':
+                    # If there are any leadership stages, convert them to a list so that the database
```

Comment:
This is the perfect amount of commenting -- they help a reader understand the code more quickly, but they're not so numerous or so verbose as to distract.

Reply:
Totally copy/pasted comments :)